idrstream/idr_stream.py

```diff
@@ -436,23 +542,49 @@ def run_stream(
-                self.compile_DP_batch_index_csv(batch_metadata)
-                # profile batch with Deep Profiler
-                self.profile_batch()
-                features_path = pathlib.Path(
-                    f"{self.final_data_dir}/batch_{batch_num}.csv.gz"
-                )
-                # compile and save features with PyCytominer
-                self.compile_batch_features(features_path)
-                # delete image/segmentation data for batch
-                self.clear_batch()
-            except Exception as e:
-                self.logger.info(f"Error while profiling batch {batch_num}:")
-                self.logger.error(e)
+            if stream_type == "CP":
+                # stream_type = "CP"
+                self.logger.info(f"Profiling batch {batch_num} with CellProfiler")
+                try:
+                    self.prepare_batch(
+                        stream_type, batch_metadata
+                    )  # put image and location data in CP-required locations
+                    self.profile_batch_with_CP()  # profile batch with CellProfiler
+                    output_path = pathlib.Path(
+                        f"{self.final_data_dir}/batch_{batch_num}.csv.gz"
+                    )
+                    self.compile_batch_CP_features(
+                        output_path
+                    )  # compile and save features with PyCytominer
+                    self.clear_batch(
+                        stream_type
+                    )  # delete image/segmentation data for batch
+                except Exception as e:
+                    self.logger.info(f"Error while profiling batch {batch_num}:")
+                    self.logger.error(e)
+
+            if stream_type == "DP":
+                # stream_type = "DP"
+                self.logger.info(f"Profiling batch {batch_num} with DeepProfiler")
+                try:
+                    self.prepare_batch(
+                        stream_type, batch_metadata
+                    )  # put image and location data in DP-required locations
+                    self.compile_batch_index_csv(
+                        stream_type, batch_metadata
+                    )  # compile index csv for DeepProfiler project for the specific batch
+                    self.profile_batch_with_DP()  # profile batch with Deep Profiler
+                    features_path = pathlib.Path(
+                        f"{self.final_data_dir}/batch_{batch_num}.csv.gz"
+                    )
+                    self.compile_batch_DP_features(
+                        features_path
+                    )  # compile and save features with PyCytominer
+                    self.clear_batch(
+                        stream_type
+                    )  # delete image/segmentation data for batch
+                except Exception as e:
+                    self.logger.info(f"Error while profiling batch {batch_num}:")
+                    self.logger.error(e)
```

Comment:
Consider making this one error log message, taking advantage of the f-string and `e`.
```suggestion
                    self.logger.error(f"Error while profiling batch {batch_num}: {e}")
```

Reply:
:+1: